### PR TITLE
Nyx: Added feature to launch OFW directly from Launch Menu, a safeui option and konami code

### PR DESCRIPTION
**Pull request by didix16**
Hello @CTCaer !

My C skills are a kind of rusty but here we go. 

This is the context:

I've had to "improve" ;) a switch console from a friend's child and since he needs to keep sysNand to play online and emunand to be offline, I wanted to make him a nice lookable lauch menu that just power on it console, appears the 2 options, One for Emmunand and one for SysNand. But I wanted that the SysNand option launches OFW instead of stock. You know, I think its better to launch OFW than using SysNand stock option, less problems I think. So, since OFW is only at Home Menu and I don't wanted that him can touch something that does not know, I decided to look for if it was possible to use the homescreen=2 option on nyx.ini and then add a launch entry fro OFW. So I've seen this issue #665 and I've wondered:  Why not implement shuch functionality?

@antonwe in that Issue was right.
> I'd like to revisit this, with modchips in mind. On my switch with a modchip installed, I'd love to immediately be presented with the launch screen with OFW boot as one of the options. I can certainly bypass the modchip by holding volume buttons during power-on or press reboot->ofw in Hekate, but when I hand the switch over to someone who doesn't fully understand how things work, having OFW as a big nice launch button would be far more obvious.

So that's what I did. I've think that for those people that does not understand tech or are unware of what a bootloader is, I think this can be a great feature. It's like kid friendly if you want. So I've also added a safeui option that avoids the creation of top right corners buttons on launch screen. That way a simple user just focus on what want to boot, nothing else. 

Also, I've added a konami code that resets the Nyx menu back to normal, just in case come person like an "admin", has the ability to access full content of hekate again and then can update it without problems. Imagine I'm the admin of my friend's child. When he needs some kind of help related with boot or needs update hekate, then I can introduce the konami code, do my stuff and put back again the "kid mode".

Sorry for my bad english. I hope the concept it is understood.

I know the code is not perfect. Just a proof of concept that is working. Just tell me what do you think and how can be adapted to the soruce code and if you are interest it (hope you are 😄  )

BTW, I've added the joycons BUTONS MASK, you know, for konami code are very handy.

### CODE REVIEW

**Comment by antonwe** [comment]
this is exactly what I wanted, thank you!

**Comment by CostelaCNX** [comment]
It's awesome trick, but only for who have switch with modchip, if you use this on Unpatched Unit, console won't boot :(

**Comment by impeeza** [comment]
There is NO WAY to run Hekate in a patched console without a modchip.

you can only run hekate on an unpatched console or a new console with modchip.

**Comment by CostelaCNX** [comment]
> There is NO WAY to run Hekate in a patched console without a modchip.
> 
> 
> 
> you can only run hekate on an unpatched console or a new console with modchip.

I know that, I say this custom setup "OFW=1" won't work if you have a unpatched console with autorcm, only for consoles with modchip. If you have autorcm on, black screen because boot0 is corrupted.

**Comment by didix16** [comment]
So @CostelaCNX , 
The actual boot OFW option from UI on bottom side, does not work either? Because I use that exact behaviour. 

I don't have AutoRCM activated and I couldn't check it. I have and unpatched unit but I prefer boot hekate  injecting the payload using a mobile with OTG cable and the jig switch.

**Comment by CTCaer** [comment]
Thanks for the PR.
I said many times why I don't want OFW there. You even read one of these.
Also any dumbing down of the UI is not something valid when anyone can just edit the files.
Anything like that will just be a google search away on how to remove, as always.
(You also touched many things in code that you shouldn't have)

I will add a reference label for anyone that actually wants to compile this.

**Comment by Fr4nk021** [comment]
@CTCaer It’s not about how easy it is to Google or edit a few files to bypass the password, nor is it about dumbing down the UI. The point is simply to make it clear that this section isn’t meant to be accessed or changed accidentally, especially by users who were handed a modded Switch and have no technical knowledge, like younger siblings. They likely won’t bother Googling the password. The goal is just to provide a safer UI, limited to selecting between CFW and OFW, so the system can be used without fear of accidentally breaking something.
@didix16 proof of concept paired with a modchip (not sure if it works with AutoRCM) that boots directly into the Safe UI with an OFW launch option sounds like a great idea, in my opinion.

**Comment by CTCaer** [comment]
Pretty doubt they won't look about it.
Also the execution of it went completely into the wrong path.
This is better suited to be paired with the password thing.

**Comment by didix16** [comment]
What @Fr4nk021  says is exactly what I wanted to express.

I have been reflecting a bit on this matter.

@CTCaer, I have seen that this is not the first time that users request you this functionality and I completely understand your point of view of not implementing it. However, I have also been able to observe that over time more and more people are realizing that this is a valuable option.

Sometimes it is assumed that all users have the same technical profile, but the truth is that there are plenty of them: apart from tech-savvy users, there is also people that, for different reasons, have a harder time using technology or are not interested in it, surprising as it may seem.

I would like to highlight the comment that @CoderQCYT included in PR #1077 describing a reality that many of us face in our immediate environment: not everyone wants or knows how to handle files, SD cards or payloads. For many people, these are barriers that may prevent them from trying because they are afraid of messing something up.

I’m quoting a sentence I think it might be relevant from @CoderQCYT on same PR #1077:

> “Could you just throw an axe through the front door? Yes, you could. Would most people do that if they wanted to get in? No, they wouldn't.”

And I’m complementing it with a similar analogy: **Anyone could look up on the Internet how to make a homemade bomb, but this doesn’t mean everyone is going to do it**. Ability doesn’t mean action and, far less, intention.

Another fact I would like to share is that you have created this bootloader. Although anyone with enough knowledge on the field could try to replicate it, the truth is that there are not many public projects like yours, and this tells a great deal about your commitment and ability. This is why I’m surprised that, having dedicated so much time and effort in making it accessible and useful, there is a certain amount of resistance in incorporating something that would make it even easier to handle for users with less technical knowledge. Wanting to help the community, but only helping those who fit on a specific profile may seem contradictory sometimes.

For instance, in my immediate environment there are more than 20 users that have taken advantage of this amendment and are grateful for it **because it allows them to have fun in a simple and safe way, without the need of dealing with technical stuff**. Not everyone has friends who can help them, nor are curious teenagers. There are parents, kids and even adults that feel insecure when dealing with these matters. It is not a problem of ability; it is a problem of context.

As for all the comments about the amendments in the code, I understand that it may not have been a perfect implementation. I acknowledge that I didn’t have much time to do it and that I did it as a proof of concept. However, when you say “You also touched many things in code that you shouldn't have”, it would be very helpful to receive a more specific explanation. **When an error is pointed out to me, I really appreciate receiving arguments and examples that can help me learn and improve**. Otherwise, the comment may be a bit ambiguous and difficult to tackle.

Moreover, I haven’t found the contributing guidelines for the project (e.g., CONTRIBUTING.md), so I assumed I could contribute in good faith following the standard PR procedure. I apologize if I did something it wasn’t expected.

In short, I just intend to bring a different perspective. I hope that instead of being seen as criticism, this message is understood as an invitation to reflect on the actual needs of users that have a less technical profile. I truly appreciate your work and the time you spend keeping it alive.

Best regards.
